### DISCUSSION TITLE
Antlr4 C# WPF build problem

### DISCUSSION THREAD

**Discussion by skulma01** [antlr/antlr4#4371] [Q&A] [answered]
Hello,

I have a hard time building ANTLR4. I tried to create a simple language just to get used to ANTLR4 but with no success.
1. Added ANTLR4 and ANTLR4.RUNTIME in nuget package
2. Created grammar and build with succes
3. Created class CalculatorVisitor
4. Build without success  - error CS0246
I don't know why. I have this problem in WPF application but not in console application. 
![3](https://github.com/antlr/antlr4/assets/99803721/15237baa-df15-4dfc-8105-933825b372a4)
![2](https://github.com/antlr/antlr4/assets/99803721/22607860-d1ec-4608-aafc-cfc8fa5469b3)
![1](https://github.com/antlr/antlr4/assets/99803721/1e6baf6d-5b38-4428-b43f-aa491157d5c6)

**Comment by kaby76** [comment, marked as answer]
The compilation error could be caused by a number of issues. I would check that the `package` statement is the same between `MainWindow.xaml.cs` and the generated files for the visitor `ArithmeticBaseVisitor.cs `and parser `ArithmeticGrammarParser.cs`. In the future, please post the code since pictures of the IDE do not show the entire code, including the .csproj. Also, you should not be using [antlr4](https://www.nuget.org/packages/Antlr4) and [antlr4.runtime](https://www.nuget.org/packages/Antlr4.Runtime). These packages are five years old, and unsupported. The official Antlr package to use is [antlr4.runtime.standard](https://www.nuget.org/packages/Antlr4.Runtime.Standard). To do the build, you should add in my package [antlr4buildtasks](https://www.nuget.org/packages/Antlr4BuildTasks), a drop-in replacement for the build rules for [antlr4](https://www.nuget.org/packages/Antlr4). Antlr4BuildTasks fixes many bugs in [antlr4](https://www.nuget.org/packages/Antlr4).

**Reply by skulma01** [reply]
Thank you very much. Tried the official Antlr with your package and it works fine.
